stix_shifter_modules/nozomi/stix_translation/transformers.py

```diff
@@ -0,0 +1,135 @@
+
+
+class ValidateMacAddr(ValueTransformer):
+    """Regex to check valid MAC address
+    Example:
+        Input: Invalid mac address
+        Output: None
+    """
+    @staticmethod
+    def transform(obj):
+        try:
+            pattern = re.compile(r'^(([0-9a-fA-F]{2}[:-]){5}([0-9a-fA-F]{2})|([0-9a-fA-F]{3}[\.]){3}([0-9a-fA-F]{3}))$')
+            if re.search(pattern, str(obj)):
+                return obj
+            return None
+        except ValueError:
+            LOGGER.error("%s connector error, cannot validate mac address: %s", connector, obj)
+            raise
+
+
+class ToFindingType(ValueTransformer):
+    """A value transformer for expected finding_type value
+    Example:
+        Input: empty string or other than empty string
+        Output: alert or threat
+    """
+    @staticmethod
+    def transform(obj):
+        try:
+            if obj in ("", "alert"):
+                return "alert"
+            return "threat"
+        except ValueError:
+            LOGGER.error("%s connector error, cannot convert input to finding type : %s", connector, obj)
+            raise
+
+
+class ToSeverityValue(ValueTransformer):
+    """A value transformer for converting severity value
+    Example:
+        Input: 9.0 or 9 or '9.0'
+        Output: 90
+    """
+    @staticmethod
+    def transform(obj):
+        try:
+            if isinstance(obj, (float, int)):
+                return int(obj * 10)
+            if obj.isnumeric():
+                return int(obj) * 10
+            if obj.replace(".", "").isnumeric():
+                return int(float(obj) * 10)
+            return None
+        except ValueError:
+            LOGGER.error("%s connector error, cannot convert input : %s", connector, obj)
+            raise
+
+
+class SizeToInteger(ValueTransformer):
+    """A value transformer for converting value to integer value
+    Example:
+        Input: '19261 bytes'
+        Output: 19261
+    """
+    @staticmethod
+    def transform(obj):
+        try:
+            if obj:
+                return int(str(obj).replace(" bytes", ""))
+            return None
+        except ValueError:
+            LOGGER.error("%s connector error, cannot convert input : %s", connector, obj)
+            raise
+
+
+class ToProtocolValue(ValueTransformer):
+    """A value transformer for converting value to protocol value
+    Example:
+        Input: 'tcp/3389' or 'unknown'
+        Output: [tcp] or [tcp]
+    """
+    @staticmethod
+    def transform(obj):
+        try:
+            if obj:
+                if '/' in obj:
+                    protocol = obj.split("/")[0]
+                    return [protocol]
+                if obj == 'unknown':
+                    return ['tcp']
+                return [obj.lower()]
+            return None
+        except ValueError:
+            LOGGER.error("%s connector error, cannot convert input : %s", connector, obj)
+            raise
+
+
+class EpochToTimestampConversion(ValueTransformer):
```

Comment:
Is there a reason why you can't use the one in stix_shiter_utils/stix_translation/src/utils/transformers.py?

Reply:
In this transformer we are checking for param value and if it is zero it returns None.
The util->transformer isn't checking for param value and it returns '1970-01-01T00:00:00.000Z' for zero.

Reply:
That seems reasonable, do you have an example where the timestamp can actually be 0?

I noticed that there is an example where the timestamp is empty in the unit test, but I don't see an example where it's 0. 

Also, why is it only applied to the "closed_time"? Wouldn't it also affect "created_time"?

One last thing, is it possible to slightly modify this to call the original transformer "EpochToTimestamp" instead of doing its own transformer? Something like

if obj:
     EpochToTimestamp.transform(obj)
and an explanation in the comments for why this isn't using the original one?

Reply:
Updated as per the suggestion.